astropy/io/fits/hdu/compressed/tests/test_tiled_compression.py

```diff
@@ -65,6 +66,103 @@ def test_zblank_support(canonical_data_base_path, tmp_path):
+    header["BLANK"] = blank
+
+    # create the compressed file
+    cfile1 = tmp_path / "compressed_with_blank.fits"
+    w = np.where(np.isnan(data))
+    blank = -32768
+    data[w] = blank
+    header["BLANK"] = blank
+    hdu = fits.CompImageHDU(
+        data=data, header=header, compression_type="RICE_1", tile_shape=(16, 16)
+    )
+    hdu.writeto(cfile1, overwrite=True, checksum=False)
+
+    # replace BLANK keyword in header with ZBLANK keyword
+    cfile2 = tmp_path / "compressed_with_zblank.fits"
+    with fits.open(cfile1, disable_image_compression=True) as hdul:
+        assert "ZBLANK" not in hdul[1].header
+        hdul[1].header["ZBLANK"] = hdul[1].header["BLANK"]
+        del hdul[1].header["BLANK"]
+        hdul.writeto(cfile2, overwrite=True, checksum=False)
+
+    # replace ZBLANK in header with with ZBLANK in table column
+    # This creates a file structure that is unlikely to be encountered in the wild
+    # but that is apparently allowed by the FITS standard.
+    cfile3 = tmp_path / "compressed_with_zblank_column.fits"
+    with fits.open(cfile2, disable_image_compression=True) as hdul:
+        phdu = hdul[0]
+        thdu = hdul[1]
+        orig_table = hdul[1].data
+        orig_cols = orig_table.columns
+        zblank = thdu.header["ZBLANK"]
+        new_cols = fits.ColDefs(
+            [
+                fits.Column(
+                    name="COMPRESSED_DATA",
+                    format="1PB()",
+                    array=thdu.data.field("COMPRESSED_DATA"),
+                ),
+                fits.Column(
+                    name="ZBLANK",
+                    format="I",
+                    array=np.zeros(len(orig_table), dtype=np.int32) + zblank,
+                ),
+            ]
+        )
+        new_thdu = fits.BinTableHDU.from_columns(new_cols, header=thdu.header)
+        del new_thdu.header["ZBLANK"]
+        new_hdul = fits.HDUList([phdu, new_thdu])
+        new_hdul.writeto(cfile3, overwrite=True, checksum=False)
+
+    # now test the 3 files to confirm they all uncompress correctly
+    for filename in (cfile1, cfile2):
+        with fits.open(canonical_data_base_path / filename) as hdul:
+            assert_equal(hdul[1].data, reference)
+            # ensure that the uncompressed header is created with BLANK keyword
+            assert hdul[1].header.get("BLANK") == -32768
+
+    # this one generates an expected warning
+    with pytest.warns(AstropyUserWarning) as record:
+        with fits.open(cfile3) as hdul:
+            assert_equal(hdul[1].data, reference)
+            assert hdul[1].header.get("BLANK") == -32768
+
+    unique_warnings = {(wm.category, wm.message.args[0]) for wm in record}
+    expected = {
+        (
+            AstropyUserWarning,
+            "Setting default value -32768 for missing BLANK keyword in compressed extension",
+        ),
+    }
+    assert (
+        unique_warnings == expected
+    ), f"Got some unexpected warnings\n{unique_warnings - expected}"
```

Comment:
All of this becomes unnecessary with my suggestion above
```suggestion
```

Reply:
This is a big improvement -- thanks for showing me the right way to do it!